app/models/manageiq/providers/terraform_enterprise/inventory/collector.rb

```diff
@@ -15,6 +15,10 @@ def workspaces
     @workspaces ||= orgs.flat_map { |org| paginated_get("organizations/#{org["id"]}/workspaces") }
   end
 
+  def workspaces_by_id
+    @workspaces_by_id = workspaces.index_by { |ws| ws["id"] }
```

Comment:
Do you need an `||=` here to avoid recalculating the index if this gets recalled?